### PR TITLE
Add a page "Examples" to the docs #232

### PR DESCRIPTION
**Pull request by vfdev-5**
Addresses #232 

- Add page mentioning examples in github 
- update links
- improve visualization of side-by-side image when click on it

### CODE REVIEW

File: docs/source/concepts.rst [hunk 1]
```diff
@@ -43,7 +43,7 @@ the run:
 - *epoch is started/completed*
 - *batch iteration is started/completed*
 
-Complete list of events can be found `here <https://github.com/pytorch/ignite/blob/master/ignite/engine/engine.py#L13>`_.
+Complete list of events can be found at :class:`ignite.engine.Events`.
```
Comment (alykhantejani) [inline docs/source/concepts.rst]: we should keep the link no?

File: docs/source/examples.rst [hunk 2]
```diff
@@ -0,0 +1,15 @@
+Examples
+========
+
+There are several examples ported from `pytorch/examples <https://github.com/pytorch/examples>`_ using `ignite`
+to display how it helps to write compact and full-featured training loops in a few lines of code:
+
+- Mnist
```
Comment (alykhantejani) [inline docs/source/examples.rst]: maybe add individual links here - will be a nicer user experience

File: docs/source/concepts.rst (hunk 1 above)
Comment (vfdev-5) [reply docs/source/concepts.rst]: I think this is bad to keep track of the line in the repository as we could add some more code in the file and this link will become obsolete. 
This rst notation :class:`ignite.engine.Events` creates a link to the doc like https://pytorch.org/ignite/engine.html#ignite.engine.Events

File: docs/source/examples.rst (hunk 2 above)
Comment (vfdev-5) [reply docs/source/examples.rst]: Good point! Will add individual links to the examples

File: docs/source/concepts.rst (hunk 1 above)
Comment (alykhantejani) [reply docs/source/concepts.rst]: ah ok that's  perfect then

**Pull request merged**